File: ios/Runner/HapticChannel.swift
import Flutter

class HapticChannel {
    private let hapticManager = HapticManager()
    private var channel: FlutterMethodChannel?
    
    func register(with messenger: FlutterBinaryMessenger) {
        channel = FlutterMethodChannel(
            name: "haptic_channel",
            binaryMessenger: messenger
        )
        channel?.setMethodCallHandler { call, result in
            switch call.method {
            case "siren": self.hapticManager.playSirenHaptic()
            case "horn":  self.hapticManager.playHornHaptic()
            case "brake": self.hapticManager.playBrakeHaptic()
            case "name":  self.hapticManager.playNameCalledHaptic()
            default: break
            }
            result(nil)
        }
    }
}

File: android/app/src/main/kotlin/com/example/eartips/HapticChannel.kt
package com.example.eartips

import android.content.Context
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodChannel

class HapticChannel(private val context: Context) {

    private val hapticManager = HapticManager(context)

    fun register(flutterEngine: FlutterEngine) {
        MethodChannel(
            flutterEngine.dartExecutor.binaryMessenger,
            "haptic_channel"
        ).setMethodCallHandler { call, result ->
            when (call.method) {
                "siren" -> {
                    hapticManager.playSiren()
                    result.success(null)
                }
                "horn" -> {
                    hapticManager.playHorn()
                    result.success(null)
                }
                "brake" -> {
                    hapticManager.playBrake()
                    result.success(null)
                }
                "name" -> {
                    hapticManager.playNameCalled()
                    result.success(null)
                }
                "cancel" -> {
                    hapticManager.cancel()
                    result.success(null)
                }
                else -> result.notImplemented()
            }
        }
    }
}